controllers/ptpconfig_controller.go

```diff
@@ -112,7 +127,325 @@ func (r *PtpConfigReconciler) syncPtpConfig(ctx context.Context, ptpConfigList *
+		for _, profile := range cfg.Spec.Profile {
+			profileName := "unknown"
+			if profile.Name != nil {
+				profileName = *profile.Name
+			}
+
+			// Check for PtpSecretName field
+			if profile.PtpSecretName != nil && *profile.PtpSecretName != "" {
+				secretName := *profile.PtpSecretName
+				glog.Infof("Found PtpSecretName in profile %s: %s", profileName, secretName)
+				if chosenSecretName == "" {
+					chosenSecretName = secretName
+				}
+			}
+
+			// Check ptp4lConf for sa_file paths
+			if profile.Ptp4lConf != nil {
+				if sa, ok := extractSaFileFromPtp4lConf(*profile.Ptp4lConf); ok {
+					if _, exists := seen[sa]; !exists {
+						seen[sa] = struct{}{}
+						saFilePaths = append(saFilePaths, sa)
+						glog.Infof("Found sa_file from PtpConfig %s, profile %s: %s", cfg.Name, profileName, sa)
+					}
+				}
+			}
+		}
+	}
+
+	// 2. If no secret name or sa_file paths, nothing to mount
+	if chosenSecretName == "" || len(saFilePaths) == 0 {
+		glog.Info("No PtpSecretName or sa_file paths found - skipping DaemonSet update")
+		return nil
+	}
+
+	// 3. Load the secret to get its keys
+	sec := &corev1.Secret{}
+	if err := r.APIReader.Get(ctx, types.NamespacedName{Namespace: names.Namespace, Name: chosenSecretName}, sec); err != nil {
+		glog.Errorf("Failed to load secret %s: %v", chosenSecretName, err)
+		return nil // Don't fail reconciliation, just skip mounting
+	}
+
+	if len(sec.Data) == 1 {
+		for k := range sec.Data {
+			chosenSecretKey = k
+			chosenSecretKeys = map[string]struct{}{k: {}}
+			glog.Infof("Found PTP security secret '%s' with key '%s'", chosenSecretName, k)
+		}
+	} else if len(sec.Data) > 1 {
+		for k := range sec.Data {
+			chosenSecretKeys[k] = struct{}{}
+		}
+		for k := range sec.Data {
+			chosenSecretKey = k
+			glog.Infof("Found PTP security secret '%s' with multiple keys, using '%s'", chosenSecretName, k)
+			break
+		}
+	}
+
+	if chosenSecretKey == "" {
+		glog.Warningf("Secret %s found but has no data keys", chosenSecretName)
+		return nil
+	}
+
+	// 4. Get the linuxptp-daemon DaemonSet
+	daemonSet := &appsv1.DaemonSet{}
+	err := r.Get(ctx, types.NamespacedName{
+		Namespace: names.Namespace,
+		Name:      "linuxptp-daemon",
+	}, daemonSet)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			glog.Info("linuxptp-daemon DaemonSet not found yet - will update on next reconcile")
+			return nil
+		}
+		return fmt.Errorf("failed to get linuxptp-daemon DaemonSet: %v", err)
+	}
+
+	// 5. Compute hash of secret data for change detection
+	secretHash := computeSecretHash(sec)
+	glog.Infof("Computed secret hash for '%s': %s", chosenSecretName, secretHash)
+
+	// 6. Inject the security volume and mounts
+	glog.Infof("Updating linuxptp-daemon DaemonSet with secret '%s' at paths: %v", chosenSecretName, saFilePaths)
+	injectPtpSecurityVolume(daemonSet, chosenSecretName, chosenSecretKey, chosenSecretKeys, saFilePaths, secretHash)
+
+	// 7. Update the DaemonSet
+	if err := r.Update(ctx, daemonSet); err != nil {
```

Comment:
Instead of updating the daemonset directly, use the merge function MergeObjectForUpdate which is called by apply.ApplyObject as in: 
[link](https://github.com/k8snetworkplumbingwg/ptp-operator/blob/81c95fdc8d4dc4b8f70ed31f3b489aa32d4700e2/controllers/ptpoperatorconfig_controller.go#L281)